tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -171,30 +176,61 @@ def argv_from_source_snapshot(self, snapshot):
     return (self.bin_path,) + tuple(snapshot_file_paths)
 
 
-# TODO: make this contain the snapshot(s?) of the output files (or contain
-# something that contains it) once we've made it so processes can make snapshots
-# of the files they produce.
-class JavacCompileResult(object): pass
+@rule(ExecuteProcessRequest, [Select(JavacCompileRequest)])
+def javac_compile_sources_execute_process_request(javac_compile_req):
+  files_to_compile = javac_compile_req.javac_sources.java_files
+
+  path_globs = PathGlobs.create('', include=files_to_compile)
+  sources_snapshot = yield Get(Snapshot, PathGlobs, path_globs)
+
+  output_files = tuple(
```

Comment:
Is this intentionally exercising multiple outputs? Otherwise you'd want to send these into a subdirectory, and then capture the subdirectory, in order to avoid missing classes generated by annotation processors, etc.

Reply:
Good point. For now, I've just added a comment that this isn't a fully featured rule and we should delete it at some point.

I think it's useful for now to act as documentation for how rules which execute processes can look, but doesn't provide much standalone value other than that (`test_write_file` tests that the API actually works), and we should just delete it when we have something more real which can act as that documentation.